SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestNetReceive/SpatialTestNetReceive.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestNetReceive.h"
+
+#include "SpatialGDK/Public/EngineClasses/SpatialNetDriver.h"
+
+#include "Net/UnrealNetwork.h"
+
+/**
+ * TODO CHANGE
+ *
+ * This test tests RepNotifies and shadow data implementation.
+ *
+ * The test includes 1 Server and 2 Client workers.
+ * The flow is as follows:
+ * - Setup:
+ *	- The test itself is used as the test Actor.
+ *  - It contains a number of replicated properties and OnRep functions that either set boolean flags confirming they have been called, or
+ *save the old value of the argument passed to the OnRep function.
+ * We also check the ordering of RepNotifies being called, expecting that both an actor and its subobjects will have their new data applied
+ *before repnotifies are called on either.
+ *
+ * - Test:
+ *	- The Server changes the values of the replicated properties multiple times.
+ *  - The clients check whether the RepNotifies have or have not been called correctly.
+ * - Cleanup:
+ *	- None.
+ */
+
+static constexpr float StepTimeLimit = 15.0f;
+ASpatialTestNetReceive::ASpatialTestNetReceive()
+	: Super()
+{
+	Author = "Arthur";
+	Description = TEXT("Test Pre/PostNetReceive events");
+}
+
+void ASpatialTestNetReceive::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("SpatialTestNetReceiveInitialise"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		TestActor = GetWorld()->SpawnActor<ASpatialTestNetReceiveActor>(Server1Pos, FRotator::ZeroRotator, FActorSpawnParameters());
+		TestActor->SetReplicates(true);
+
+		if (!AssertIsValid(TestActor, TEXT("Failed to spawn TestActor")))
```

Comment:
Subtle point, but we are not really testing the ability to spawn TestActors, would use and `check` instead.

Reply:
added :)